tests/functional/test_s3.py

```diff
@@ -423,3 +423,21 @@ def test_region_redirect_cache(self):
                      '?encoding-type=url')
         self.assertEqual(calls[1].url, fixed_url)
         self.assertEqual(calls[2].url, fixed_url)
+
+
+class TestPresign(BaseS3OperationTest):
```

Comment:
We should also check these cases for presigned posts as they are a completely different auth identifier.